File: Семестр 1/ДЗ 6/задача_1/list.cpp
#include "list.h"
#include <iostream>
#include <string>;

List *createList()
{
	List *list = new List;
	list->head = new ListElement;
	list->head->next = NULL;
	list->head->value = NULL; 
	return list;
}

int sizeOfList(List* list)
{
	int size = 0;
	ListElement *tmp = list->head->next;
	while (tmp != NULL)
	{
		++size;
		tmp = tmp->next;
	}
	return size;
}

Position elemPos(ListType value, List *list)
{
	ListElement* tmp = list->head;
	while(tmp->value != value)
		tmp=tmp->next;
	return tmp;
}

void addElem(Position p, List *list, ListType value)
{
	ListElement* tmp = new ListElement;
	tmp->value = value;
	tmp->next = p->next;
	p->next = tmp;
}

void deleteElem(ListType value, List *list)
{
	ListElement* tmp = list->head;
	while (tmp->next->value != value)
		tmp = tmp->next;
	ListElement *tmp1 = elemPos(value, list);
	tmp->next = tmp1->next;
    delete tmp1;
}

void printList(List *list)
{
	ListElement* tmp = list->head;
	while (tmp->next != NULL)
	{
		cout << tmp->next->value->str << " " << tmp->next->value->num << endl;
		tmp = tmp->next;
	}
}

void deleteList(List *list)
{
	ListElement *tmp = list->head;
	ListElement *tmp1 = tmp->next;
	while (tmp1 != NULL)
	{
		tmp = tmp1;
		tmp1 = tmp1->next;
		delete tmp;
	}
	delete list->head;
	delete list;
}

File: Семестр 2/ДЗ 3/problem_2/problem_2/List.cs
using System;

namespace problem_2
{
    public class List
    {
        public class ListElement
        {
            private int number;
            private ListElement next;
            public int Number
            {
                get 
                {
                    return number; 
                }
                set
                {
                    number = value;
                }
            }
            public ListElement Next
            {
                get
                {
                    return next;
                }
                set
                {
                    next = value;
                }
            }
        }
        private ListElement head;
        public ListElement Head
        {
            get
            {
                return head;
            }
            private set
            {
                head = value;
            }
        }

        public List()
        {
            Head = new ListElement();
            Head.Next = null;
            Head.Number = 0;
        }

        public int SizeOfList()
        {
            int size = 0;
            ListElement tmp = head.Next;
            while (tmp != null)
            {
                ++size;
                tmp = tmp.Next;
            }
            return size;
        }

        public ListElement ElemPos(int value)
        {
            ListElement tmp = head;
            while (tmp.Number != value)
                tmp = tmp.Next;
            return tmp;
        }

        public void AddElem(ListElement pos, int value)
        {
	        ListElement tmp = new ListElement();
	        tmp.Number = value;
	        tmp.Next = pos.Next;
	        pos.Next = tmp;
        }

        public void DeleteElem(int value)
        {
            ListElement tmp = head;
            while (tmp.Next.Number != value)
            {
                tmp = tmp.Next;
                if (tmp.Next == null)
                    break;
            }
            if (tmp.Next != null)
            {
                ListElement tmp1 = ElemPos(value);
                tmp.Next = tmp1.Next;  
            }          
        }

        public void PrintList()
        {
            ListElement tmp = head;
            while (tmp.Next != null)
            {
                Console.Write("{0} ", tmp.Next.Number);
                tmp = tmp.Next;
            }
            Console.WriteLine();
        }
    }
}
